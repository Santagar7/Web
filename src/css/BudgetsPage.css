.budgets_page{
    width: 100%;
    height:100%;
    position: absolute;
    top: 0;
    text-align: center;
}
.topnav {
    overflow: hidden;
    background-color: #333;
    width:100%;
    height:55px;
    position: top;
}
.topnav a {
    float: right;
    color: #f2f2f2;
    text-align: center;
    padding: 14px 16px;
    text-decoration: none;
    font-size: 24px;
    height:55px;
    cursor: pointer;
}
.topnav a:hover {
    background-color: #ddd;
    color: black;
}
.topnav a.active {
    background-color: #04AA6D;
    color: white;
}
.budgets{
    overflow-y: scroll;
    padding: 20px;
    box-sizing: border-box;
    max-height: 240px;
}
a {
    all: unset;
}
.budgets a{
    width: 100%;
    display: block;
    text-align: center;
    padding: 20px;
    box-sizing: border-box;
    text-decoration: none;
    font-size: 24px;
    cursor: pointer;
}
.budgets a:hover{
    background: #ddd;
}
.creator{
    width: 50%;
    margin: 10px 25%;
    padding: 10px;
    box-sizing: border-box;
    border-radius: 10px;
    border: solid silver 2px;
    cursor: pointer;
}
.not_active{
    display:none;
}
button {
    width: 50%;
    height: 30px;
    margin-top: 20px;
    border: 1px solid silver;
    background: HoneyDew;
    border-radius: 25px;
    font-size: 16px;
    color: blue;
    font-weight: 700;
    cursor: pointer;
    outline: none;
}
button:hover {
    border-color: blue;
    transition: 0.4s;
}
.create_budget{
    position:absolute;
    left: 0;
    right: 0;
    margin: 0 auto;
}
.create_budget input{
    width: 50% !important;
    margin: 15px 25%;
    padding: 10px;
    box-sizing: border-box;
    transition: 0.2s ease all;
    border:none;
    border-bottom: 2px solid silver;
    background: whitesmoke;
    font-size: 24px;
}
.create_budget input:focus{
    outline: none;
    border-bottom: 2px solid blue;
}
.create_budget input:hover {
  border-bottom: 2px solid blue;
}