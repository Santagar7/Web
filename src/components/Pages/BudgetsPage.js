import '../../css/BudgetsPage.css'
import { Link } from 'react-router-dom'
import React, { useState, useEffect } from 'react'
import axios from 'axios'

import Navbar from '../Elements/Navbar'
import CreateBudgetPanel from '../Elements/CreateBudgetPanel'

const url = 'http://localhost:7000/'

function BudgetsPage () {
  const [active, setShowPanel] = useState(false)
  const [budgets, setBudgets] = useState([])
  useEffect(() => {
    axios.get(url + 'my_budgets').then((resp) => {
      setBudgets(resp.data.data || [])
    })
  }, [])
  return (
        <div className="budgets_page">
            <CreateBudgetPanel active={active} func={setShowPanel}/>
            <Navbar/>
            <div className={!active ? 'center' : 'center not_active'}>
                <h1>My budgets</h1>
                <div className="budgets" id="budgets">
                    {budgets.map(item =>
                      (<Link key={item.index} to={'/menu/' + item.index}>{item.name}</Link>))}
                </div>
                <button onClick={() => setShowPanel(true)}>Create new budget</button>
            </div>
        </div>
  )
}

export default BudgetsPage
