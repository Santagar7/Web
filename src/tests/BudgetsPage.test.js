import { BrowserRouter } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import BudgetsPage from '../components/Pages/BudgetsPage';
import renderer from 'react-test-renderer';

test('should render correctly', () => {
    const tree = renderer.create(<BrowserRouter><BudgetsPage/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
test('test submit btn', () => {
    render(
        <BrowserRouter>
            <BudgetsPage />
        </BrowserRouter>,
    );
    const submit = screen.getByText(/Create a budget/i);
    fireEvent.click(submit);
});
test('test text', () => {
    render(
        <BrowserRouter>
            <BudgetsPage />
        </BrowserRouter>,
    );
    const elem = screen.getByText(/My budgets/i);
    expect(elem).toBeInTheDocument();
});