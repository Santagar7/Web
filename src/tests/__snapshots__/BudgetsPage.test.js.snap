// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="budgets_page"
>
  <div
    className="create_budget not_active"
  >
    <div
      className="center"
    >
      <h1
        id="message"
      >
        Create new budget
      </h1>
      <form>
        <input
          data-testid="name"
          id="name"
          required={true}
          type="text"
        />
        <button
          onClick={[Function]}
        >
          Submit
        </button>
      </form>
      <button>
        Cancel
      </button>
    </div>
  </div>
  <div
    className="extra"
  >
    <div
      className="topnav"
    >
      <a
        className="active"
        onClick={[Function]}
      >
        Log Out
      </a>
      <a
        data-testid="members"
        href="/"
        id="members"
        onClick={[Function]}
      >
        Members
      </a>
      null:
      <a
        data-testid="budgets"
        href="/budgets"
        onClick={[Function]}
      >
        Budgets
      </a>
      null:
      <a
        data-testid="action"
        href="/"
        id="action"
        onClick={[Function]}
      >
        Actions
      </a>
    </div>
  </div>
  <div
    className="center"
  >
    <h1>
      My budgets
    </h1>
    <div
      className="budgets"
      id="budgets"
    />
    <button
      onClick={[Function]}
    >
      Create a budget
    </button>
  </div>
</div>
`;
